lib/charcoal/jsonp.rb

```diff
@@ -38,8 +42,8 @@ def add_jsonp_callback
       yield
 
       if response.status.to_s.starts_with?('200') && jsonp_request?
-        response.body = "#{params[:callback]}(#{response.body})"
         response.content_type = "application/javascript"
+        response.body = "#{params[:callback]}(#{response.body.to_s})"
       end
     end
```

Comment:
Calling `response.body` will freeze the response. 

See https://github.com/rails/rails/blob/master/actionpack/lib/action_dispatch/http/response.rb#L42 and https://github.com/rails/rails/blob/master/actionpack/lib/action_dispatch/http/response.rb#L114-L116

This is a bit ugly and I'm totally unsure of the result with stream-based responses... Any suggestion?